pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -1032,7 +1032,8 @@ cdef class PyObjectHashTable(HashTable):
             val = values[i]
             hash(val)
 
-            if ignore_na and ((val != val or val is None)
+            if ignore_na and (((val == val) is (val != val))
```

Comment:
I think would be clearer if you just added `or val is pd.NA`

Reply:
Are we able to do that here? I'm not too familiar with `pxi.in` files

Reply:
Yea import machinery should still be the same

Reply:
Ok yeah that's way nicer